### PR TITLE
Added AppImage structure

### PR DESCRIPTION
**Pull request by djrm**
This will allow the distribution of AppImages for linux

### CODE REVIEW

File: tools/Godot.AppDir/godot.desktop [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[Desktop Entry]
+Name=Godot Engine
+Exec=godot %u
+Icon=godot
+Categories=Game development;Game engine;Games;Development
```
Comment (akien-mga) [inline tools/Godot.AppDir/godot.desktop]: Those are not valid FreeDesktop categories.
I use this for Mageia:
```
[Desktop Entry]
Name=Godot Engine
GenericName=Libre game engine
Comment=Debug information for package godot
Exec=godot -pm
Icon=godot
Terminal=false
Type=Application
Categories=Development;IDE;
```

File: tools/Godot.AppDir/AppRun [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+HERE=$(dirname $(readlink -f "${0}"))
+export LD_LIBRARY_PATH="${HERE}"/usr/lib:$PATH
+"${HERE}"/usr/bin/godot $@
```
Comment (akien-mga) [inline tools/Godot.AppDir/AppRun]: This is a bit convoluted as we don't need a `LD_LIBRARY_PATH`, I think you could just drop the `godot` binary in the main folder and run it directly.

**Comment by akien-mga** [comment]
We really need to cleanup `tools/`, it's getting incredibly crowded :/

**Comment by djrm** [comment]
Yes, you are right, i put the AppDir folder in tools becase i saw other folders like it there.

**Comment by reduz** [comment]
yeah, i suppose some of the osx/ios stuff could go to their respective
platforms

**Comment by akien-mga** [comment]
I'll move stuff around, and then ask you to rebase accordingly :)

**Comment by djrm** [comment]
ok

**Comment by reduz** [comment]
feel free to merge if you find it works

**Comment by akien-mga** [comment]
@djrm I finally did the `tools` cleanup I wanted, so if my PR (#6364) is merged in its current state, I'll ask you to move your appimage structure to `tools/dist`. But we'll see if others agree with this path first.

File: tools/Godot.AppDir/AppRun [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+HERE=$(dirname $(readlink -f "${0}"))
```
Comment (procedural) [inline tools/Godot.AppDir/AppRun]: Always be careful with bash, this doesn't work for folders with spaces:

image

The correct command:

```bash
HERE="$(dirname "$(readlink -f "${0}")")"
```

**Comment by djrm** [comment]
@akien-mga ok, i will move this once your PR gets merged, @procedural, ok i will change before pushing this again.

**Comment by akien-mga** [comment]
@djrm Should be good to rebase now :) Maybe rename the folder to something a bit clearer too (see the other examples in `tools/dist`, mentioning that it's Godot is maybe not necessary in the folder name).

**Comment by djrm** [comment]
ok, will check that out

**Comment by akien-mga** [comment]
Merged manually in ef174ab.

**Pull request closed without merging**